File: Chicken.php
<?php
spl_autoload_register(function ($class_name) {
    include $class_name . '.php';
});

class Chicken extends Animal
{
    public $animalId;
    public $animalBreed = 'chickens';
    public $product = 'eggs';

    function __construct()
    {
        $this->animalId = parent::$id++;
    }

    public function getOutputProduct()
    {
        return rand(0, 1);
    }
}

File: go/animals/Chicken.go
package animals

import "math/rand"

type Chicken struct {
	AnimalBreed, Product string
}

func (chicken *Chicken) Init() Chicken {
	return Chicken{
		AnimalBreed: "chickens",
		Product: "eggs",
	}
}

// Returns a random amount of products.
func (chicken *Chicken) GetOutputProduct() int {
	return rand.Intn(2)
}
